src/idc/idc.c

```diff
@@ -126,6 +126,40 @@ static void idc_ipc(void)
 	ipc_cmd(ipc->comp_data);
 }
 
+#if CONFIG_IPC_MAJOR_4
+static int idc_ipc4_bind(uint32_t comp_id)
+{
+	struct ipc_comp_dev *ipc_dev;
+	struct idc_payload *payload;
+	struct ipc4_module_bind_unbind *bu;
+
+	ipc_dev = ipc_get_comp_by_id(ipc_get(), comp_id);
+	if (!ipc_dev)
+		return -ENODEV;
+
+	payload = idc_payload_get(*idc_get(), cpu_get_id());
```

Comment:
what heppens if payload is NULL ? best to check.

Reply:
idc_payload_get() returns pointer to static_payload array (+ offset depending on core). That static_payload is declared statically in idc.c and so never is NULL (unless someone modify the code).